src/core/sr.ml

```diff
@@ -141,9 +142,11 @@ let check_rule : sym StrMap.t -> sym * pp_hint * rule Pos.loc -> unit =
       fatal r.pos "Rule [%a] does not preserve typing." pp_rule (s,h,r.elt)
   | Some(cs) ->
   let is_constr c =
-    let eq_comm (t1,u1) (t2,u2) =
-      (Eval.eq_modulo t1 t2 && Eval.eq_modulo u1 u2) ||
-      (Eval.eq_modulo t1 u2 && Eval.eq_modulo t2 u1)
+    let eq_comm (_,t1,u1) (_,t2,u2) =
+      (* Contexts ignored: [solve] doesn't generate contexts with defined
```

Comment:
No, the reason is that, line 118, Typing.infer has been called with an empty context and, indeed, neither infer nor solve introduce defined variables.